datafusion-cli/src/exec.rs

```diff
@@ -192,17 +193,20 @@ async fn exec_and_print(
     let now = Instant::now();
 
     let sql = unescape_input(&sql)?;
-    let plan = ctx.state().create_logical_plan(&sql).await?;
-    let df = match &plan {
-        LogicalPlan::Ddl(DdlStatement::CreateExternalTable(cmd)) => {
-            create_external_table(ctx, cmd).await?;
-            ctx.execute_logical_plan(plan).await?
-        }
-        _ => ctx.execute_logical_plan(plan).await?,
-    };
+    let statements = DFParser::parse_sql(&sql)?;
```

Comment:
This will use the default dialect, we can use `parse_sql_with_dialect()` and pass the dialect from the current context into it.

Reply:
I am not sure what you mean by the current context -- which one are you thinking of?

Reply:
```
DataFusion CLI v28.0.0
❯ set datafusion.sql_parser.dialect = 'Postgres';
```
I'm referring to this option, it can be found in `ctx.state()...`, and be used to parse statements

Reply:
Done. Please let me know if there are any errors or omissions.

Reply:
Looks good to me, thank you!